### PR TITLE
Add support for json logging

### PR DESCRIPTION
**Pull request by lalbers**
Related to issue #47.

Enable support for configurable structured json logging.

### CODE REVIEW

File: deploy/injector-deployment.yaml [hunk 1]
```diff
@@ -32,6 +32,8 @@ spec:
               value: ":8080"
             - name: AGENT_INJECT_LOG_LEVEL
               value: "info"
+            - name: AGENT_INJECT_LOG_FORMAT
+                - value: "json"
```
Comment (jasonodonnell) [inline deploy/injector-deployment.yaml]: Format is off here.

Comment (jasonodonnell) [reply deploy/injector-deployment.yaml]: I don't think he default should be `json`.  It's nice that there's a configurable for this now, however, not everyone is going to want JSON by default.

File: subcommand/injector/flags.go [hunk 2]
```diff
@@ -25,6 +25,9 @@ type Specification struct {
 	// LogLevel is the AGENT_INJECT_LOG_LEVEL environment variable.
 	LogLevel string `split_words:"true"`
 
+	// LogFormat is the AGENT_INJECT_LOG_FORMAT environment variable
+	LogFormat string `split_words:"true"`
```
Comment (jasonodonnell) [inline subcommand/injector/flags.go]: Please add a test for this (there's an existing unit test for envs).

File: deploy/injector-deployment.yaml (hunk 1 above)
Comment (lalbers) [reply deploy/injector-deployment.yaml]: I agree, default log settings should be similar to [hashicorp/vault](https://github.com/hashicorp/vault). Fixed.

**Comment by frittentheke** [comment]
This is really great @lalbers !

Looking at the field names there is the "@" sign in front. Not that this breaks anything, but in the context of shipping stuff to an ElasticSearch this sometime needs extra treatment. Especially when looking at "@timestamp" ...

```
{"@level":"info","@message":"Request received","@module":"handler","@timestamp":"2020-01-21T08:43:25.828108Z","Method":"POST","URL":{"Scheme":"","Opaque":"","User":null,"Host":"","Path":"/mutate","RawPath":"","ForceQuery":false,"RawQuery":"timeout=30s","Fragment":""}}
{"@level":"debug","@message":"checking if should inject agent..","@module":"handler","@timestamp":"2020-01-21T08:43:25.828687Z"}
{"@level":"debug","@message":"checking namespaces..","@module":"handler","@timestamp":"2020-01-21T08:43:25.828707Z"}
{"@level":"debug","@message":"setting default annotations..","@module":"handler","@timestamp":"2020-01-21T08:43:25.828716Z"}
{"@level":"debug","@message":"creating new agent..","@module":"handler","@timestamp":"2020-01-21T08:43:25.828739Z"}
{"@level":"debug","@message":"validating agent configuration..","@module":"handler","@timestamp":"2020-01-21T08:43:25.828790Z"}
```


Is there maybe an easy way to remove those?

File: subcommand/injector/flags.go [hunk 3]
```diff
@@ -45,14 +49,15 @@ type Specification struct {
...
 	VaultAuthPath string `split_words:"true"`
-
 }
 
 func (c *Command) init() {
 	c.flagSet = flag.NewFlagSet("", flag.ContinueOnError)
 	c.flagSet.StringVar(&c.flagListen, "listen", ":8080", "Address to bind listener to.")
 	c.flagSet.StringVar(&c.flagLogLevel, "log-level", DefaultLogLevel, "Log verbosity level. Supported values "+
 		`(in order of detail) are "trace", "debug", "info", "warn", and "err".`)
+	c.flagSet.StringVar(&c.flagLogFormat, "log-level", DefaultLogFormat, "Log output format. "+
```
Comment (jasonodonnell) [inline subcommand/injector/flags.go]: Because `log-level` is redefined here (it should be `log-format`), this is causing the injector to panic.

Comment (jasonodonnell) [reply subcommand/injector/flags.go]:
```
flag redefined: log-level
panic: flag redefined: log-level

goroutine 1 [running]:
flag.(*FlagSet).Var(0xc0001ea5a0, 0x1435de0, 0xc0001fa210, 0x1291171, 0x9, 0x12cd800, 0x3d)
	/usr/local/Cellar/go/1.12.7/libexec/src/flag/flag.go:850 +0x4af
flag.(*FlagSet).StringVar(...)
	/usr/local/Cellar/go/1.12.7/libexec/src/flag/flag.go:753
github.com/hashicorp/vault-k8s/subcommand/injector.(*Command).init(0xc0001fa1e0)
	/Users/jasonodonnell/Git/vault-k8s/subcommand/injector/flags.go:59 +0x1b5
sync.(*Once).Do(0xc0001fa298, 0xc00006bdd0)
	/usr/local/Cellar/go/1.12.7/libexec/src/sync/once.go:44 +0xb3
github.com/hashicorp/vault-k8s/subcommand/injector.(*Command).Run(0xc0001fa1e0, 0xc00003a0b0, 0x1, 0x1, 0x0)
	/Users/jasonodonnell/Git/vault-k8s/subcommand/injector/command.go:53 +0xe1
github.com/mitchellh/cli.(*CLI).Run(0xc0001afe00, 0x9, 0xc0001d8400, 0x128fea8)
	/Users/jasonodonnell/go/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x1f1
main.main()
	/Users/jasonodonnell/Git/vault-k8s/main.go:17 +0x16b
```

**Comment by lalbers** [comment]
Fixed typo in flag-name

**Pull request merged**